client/blocks/signup-form/index.jsx

```diff
@@ -468,7 +469,7 @@ class SignupForm extends Component {
 				times_email_validation_failed: timesEmailValidationFailed,
 			};
 
-			this.props.submitForm( this.state.form, this.getUserData(), analyticsData, () => {
```

Comment:
Form was not being used at all, removing

Reply:
We can't simpy remove a parameter passed to a function that arrives from props. What about others that are passing a `submitForm` function that uses form?
What about something like https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/invites/p2/invite-accept-logged-out.jsx#L31.

Reply:
It would work because they are using the same function and I have removed from both. To avoid any unexpected side effects, let's keep it 👍 
Putting back.